src/shared/condition.c

```diff
@@ -958,6 +960,118 @@ static int condition_test_file_is_executable(Condition *c, char **env) {
                 (st.st_mode & 0111));
 }
 
+static int condition_test_psi(Condition *c, char **env) {
+        _cleanup_free_ char *first = NULL, *second = NULL, *pressure_path = NULL;
+        loadavg_t *current, limit;
+        ResourcePressure pressure;
+        const char *p, *value;
+        OrderOperator order;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(IN_SET(c->type, CONDITION_MEMORY_PRESSURE, CONDITION_CPU_PRESSURE));
+
+        if (!is_pressure_supported())
+                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Pressure Stall Information (PSI) is not supported");
```

Comment:
mmh, to maximize portability of unit files I'd say that on a kernel that lacks PSI we assume the system is always idle.